### PR TITLE
[Cleanup] Follow styling conventions for const/enums

### PR DESCRIPTION
**Pull request by nkuang123**
Clean up styling as per https://docs.godotengine.org/en/4.0/tutorials/scripting/gdscript/gdscript_styleguide.html

### CODE REVIEW

File: game_data.gd [hunk 1]
```diff
@@ -1,41 +1,60 @@
...
-    TETROMINO.double_square: 0
+const TILE_ID = {
+    Tetromino.SQUARE: 0,
+    Tetromino.TEE: 1,
+    Tetromino.L_LEFT: 2,
+    Tetromino.L_RIGHT: 3,
+    Tetromino.ZAG_RIGHT: 4,
+    Tetromino.ZAG_LEFT: 5,
+    Tetromino.LINE: 6,
+    Tetromino.DOUBLE_SQUARE: 0,
```
Comment (nkuang123) [inline game_data.gd:27]: @rp282 Why does DOUBLE_SQUARE have the same ID as SQUARE?

Comment (rp282) [reply game_data.gd:27]: These are the tile ids used to pick the colors of the tetrominos. We'll need to decide if our added pieces have their own tiles/

Comment (nkuang123) [reply game_data.gd:27]: I think we should have a separate list for piece indices and tile IDs (colors):
- Piece index: a unique identifier for each piece
- Tile ID: maps to the tile ID map

This way, it'll be possible for different pieces to have the same tile IDs i.e. square & double square with tile ID 0. Else we run into situations where we can't disambiguate pieces i.e. piece holding.

**Review by rp282 — APPROVED** [review]
lgtm basically threw out changes for new GameData

**Pull request merged**